hello-server.php

```diff
@@ -45,74 +43,184 @@
+	}
+
+	// Return our IP.
+	return $server_ip;
 }
 
-// This just echoes the chosen line, we'll position it later
-function hello_server( $wp_admin_bar ) {
-	$serverInfo = hello_server_get_info();
-	$mainNode = [
-	  'id' => 'lw_hello_server',
-	  'title' => __("Hello, Server", "hello-server"),
-	  'meta' => ['title' => __('View the current server info.', "hello-server"), 'class' => 'menupop']
-
-	];
-	$hostNode = [
-	  'id' => 'lw_host_server',
-      'title' => "Hello, I'm: <span itemprop='server-name'>".$serverInfo['hostname'] . "</span>!",
-	  'meta' => ['title' => 'View the current server hostname.'],
-	  'parent' => 'lw_hello_server'
-	];
-	$ipNode = [
-	  'id' => 'lw_server_ip',
-	  'title' => 'My IP is: '.$serverInfo['ip'],
-	  'meta' => ['title' => 'The current servers IP Address.'],
-	  'parent' => 'lw_hello_server'
-	];
-	$wp_admin_bar->add_node( $mainNode );
-	$wp_admin_bar->add_node( $hostNode );
-	$wp_admin_bar->add_node( $ipNode );
+/**
+ * Include a small bit of CSS to display and position the icon.
+ *
+ * @return void
+ */
+function lw_add_admin_bar_css() {
+
+	echo '
+	<style>
+
+		li#wp-admin-bar-lw-hello-server {
+			display: block;
+		}
+
+		li#wp-admin-bar-lw-hello-server .ab-item.ab-empty-item .ab-lw-icon {}
+
+		li#wp-admin-bar-lw-hello-server .ab-item.ab-empty-item .ab-lw-icon:before {
+			content: "\f115";
+			top: 2px;
+		}
+
+		li#wp-admin-bar-lw-hello-server .lw-admin-child-menu .ab-item.ab-empty-item span {
+			color: #fff;
+		}
+
+		@media screen and ( max-width: 782px ) {
+
+			li#wp-admin-bar-lw-hello-server {
+				display: block !important;
+			}
+		}
+
+	</style>
+	';
 }
+add_action( 'wp_head', 'lw_add_admin_bar_css' );
+add_action( 'admin_head', 'lw_add_admin_bar_css' );
+
+/**
+ * Display our server info in the admin bar.
+ *
+ * @param  WP_Admin_Bar $wp_admin_bar  The global WP_Admin_Bar object.
+ *
+ * @return void
+ */
+function lw_load_hello_server( WP_Admin_Bar $wp_admin_bar ) {
+
+	// Fetch our server info.
+	$info	= lw_hello_server_get_info();
 
-// Now we set that function up to execute when the admin_notices action is called
-add_action( 'admin_bar_menu', 'hello_server', 999 );
+	// Add a parent item.
+	$wp_admin_bar->add_node(
+		array(
+			'id'    => 'lw-hello-server',
+			'title' => '<span class="ab-icon ab-lw-icon"></span><span class="ab-label ab-lw-label">' . __( 'Hello Server', 'hello-server' ) . '</span>',
+			'meta'  => array(
+				'title' => __( 'View the current server info.', 'hello-server' ),
+				'class' => 'lw-admin-parent-menu',
+			),
+		)
+	);
 
-// Internationalization Support
-function load_plugins_textdomain()
-{
-	load_plugin_textdomain(
-		'hello-server',
-		false,
-		dirname(dirname(plugin_basename(__FILE__))) . '/languages/'
+	// Add the host server info.
+	$wp_admin_bar->add_node(
+		array(
+			'id'        => 'lw-host-server',
+			'title'     => sprintf( __( 'Server Name: %s', 'hello-server' ), '<span itemprop="server-name">' . esc_attr( $info['hostname'] ) . '</span>' ),
```

Comment:
`__( 'Server Name: %s', 'hello-server' )` should be late-escaped; fortunately, `esc_html__()` would be applied to the translation _before_ `sprintf()`, so your `span.server-name` should be fine.

Additionally, `$info['hostname']` should be escaped via `esc_html()`, not `esc_attr()` as it's not being used in an HTML attribute.